src/models/guild_preference.ts

```diff
@@ -223,11 +231,13 @@ export default class GuildPreference {
         return this.gameOptions.shuffleType === ShuffleType.UNIQUE;
     }
 
-    async updateGuildPreferences(_db: Knex) {
+    async updateGuildPreferences(_db: Knex, songsState: AvailableSongs) {
         await _db("guild_preferences")
             .where({ guild_id: this.guildID })
             .update({ guild_preference: JSON.stringify(this) });
         const gameSession = state.gameSessions[this.guildID];
-        if (gameSession) gameSession.resetLastPlayedSongsQueue();
+        if (gameSession && songsState === AvailableSongs.Modifies) {
+            gameSession.resetLastPlayedSongsQueue();
```

Comment:
Don't we still have this logic in gamesession

Reply:
That call to reset is when the user plays all of their unique songs, so it's still needed